experiments/collect_unsupported.cpp

```diff
@@ -0,0 +1,167 @@
+#include <iostream>
+#include <fstream>
+#include <cassert>
+#include <vector>
+#include <map>
+#include <string>
+#include <set>
+#include <algorithm>
+
+using namespace std;
+
+struct Feature
+{
+   string calling_function;
+   string message;
+   string nkind;
+   string source_file;
+
+   size_t source_location;
+   string node_id;
+
+   map<string, string> parameters;
+
+   void read_header(const string& p) {
+      if (!p.empty() && p[0] == '-' && p.size() > 20) {
+         if (p.find("At:") != string::npos)
+            calling_function = p.substr(14, p.size() - 24);
+         else
+            message = p.substr(10, p.size() - 20);
+      }
+   }
+
+   void read_param(const string& p) {
+      auto equal_sign_pos = p.find('=');
+      if (equal_sign_pos != string::npos)
+         parameters[p.substr(1, equal_sign_pos - 2)] = p.substr(equal_sign_pos + 2);
+   }
+
+   void read_nkind(const string& p) {
+      auto space_sign_pos = p.find(' ');
+      if (space_sign_pos != string::npos)
+         nkind = p.substr(0, space_sign_pos);
+      auto equal_sign_pos = p.find('=');
+      if (equal_sign_pos != string::npos) {
+         auto close_bracket_sign_pos = p.find(')', equal_sign_pos);
+         if (close_bracket_sign_pos != string::npos)
+            node_id = p.substr(equal_sign_pos + 1, (close_bracket_sign_pos - equal_sign_pos) - 1);
+      }
+   }
+
+   Feature(vector<string>& params) {
+      for (auto &p : params) {
+         switch (p[0]) {
+         case '-' :
+            read_header(p);
+            break;
+         case ' ' :
+            read_param(p);
+            break;
+         default:
+            read_nkind(p);
+            break;
+         }
+      }
+   }
+
+   // const Feature& operator=(const Feature& other) {
+
+   // }
+
+   string unique_id() const {
+      return calling_function + message + nkind;
+   }
+
+   bool operator==(const Feature& other) const {
+      return unique_id() == other.unique_id();
+   }
+
+   bool operator<(const Feature& other) const {
+      return unique_id() < other.unique_id();
+   }
+
+   string get_param(const string& name) const {
+      return parameters.at(name);
+   }
+
+   void printout(string& output) const {
```

Comment:
I'd prefer this just taking a stream instead. Then you can use stringstream if you want a string, or otherwise write directly whereever you want it to go.